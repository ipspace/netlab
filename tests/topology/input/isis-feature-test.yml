#
# Sample multi-vendor configuration that test most features supported by IS-IS configuration module
#
# Used in manual testing (feel free to write an automated test script ;)
#
addressing:
  p2p:
    unnumbered: true
  stub:
    ipv4: 172.31.0.0/16
    prefix: 24
    ipv6: 2008:db8:0001::/48

isis:
  area: '49.0002'

module: [ isis ]

nodes:
- name: c_nxos
  device: nxos
  isis:
    net: 49.0003.0000.0000.1234.00
- name: c_csr
  device: csr
  isis:
    area: '49.0003'
- name: a_eos
  device: eos
- name: j_vsrx
  device: vsrx

links:
- c_nxos:
    isis:
      metric: 10
  c_csr:
    isis:
      metric: 20
  a_eos:
    isis:
      metric: 30
  j_vsrx:
    isis:
      metric: 50

# Stub links:
- c_nxos:
- c_csr:
- a_eos:
- j_vsrx:

# Passive link (using IGP flag)
# Testing IS-IS metric specified on the link
- role: stub
  isis:
    metric: 50
    passive: True
  c_nxos:
  c_csr:
  a_eos:
  j_vsrx:

# External link - should not be in IGP
- role: external
  c_nxos:
  c_csr:
  a_eos:
  j_vsrx:

# P2P links - test unnumbered interfaces
- c_nxos:
  a_eos:

- c_nxos:
  c_csr:

# IS-IS on vSRX does not work over unnumbered IPv4 interfaces.
# All unnumbered test links for vSRX are v6-only
- c_nxos:
  j_vsrx:
  prefix:
    ipv6: True

- a_eos:
  j_vsrx:
  prefix:
    ipv6: True

- a_eos:
  c_csr:

- c_csr:
  j_vsrx:
  prefix:
    ipv6: True

- name: L2-only link to test removal of isis context
  prefix:
    ipv4: False
  c_nxos:
  a_eos:
