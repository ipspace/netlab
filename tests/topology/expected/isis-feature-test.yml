input:
- topology/input/isis-feature-test.yml
- package:topology-defaults.yml
isis:
  area: 49.0002
  type: level-2
links:
- bridge: input_1
  interfaces:
  - ipv4: 172.16.0.1/24
    isis:
      metric: 10
    node: c_nxos
  - ipv4: 172.16.0.2/24
    isis:
      metric: 20
    node: c_csr
  - ipv4: 172.16.0.3/24
    isis:
      metric: 30
    node: a_eos
  - ipv4: 172.16.0.4/24
    isis:
      metric: 50
    node: j_vsrx
  linkindex: 1
  node_count: 4
  prefix:
    ipv4: 172.16.0.0/24
  type: lan
- bridge: input_2
  interfaces:
  - ipv4: 172.16.1.1/24
    node: c_nxos
  linkindex: 2
  node_count: 1
  prefix:
    ipv4: 172.16.1.0/24
  type: stub
- bridge: input_3
  interfaces:
  - ipv4: 172.16.2.2/24
    node: c_csr
  linkindex: 3
  node_count: 1
  prefix:
    ipv4: 172.16.2.0/24
  type: stub
- bridge: input_4
  interfaces:
  - ipv4: 172.16.3.3/24
    node: a_eos
  linkindex: 4
  node_count: 1
  prefix:
    ipv4: 172.16.3.0/24
  type: stub
- bridge: input_5
  interfaces:
  - ipv4: 172.16.4.4/24
    node: j_vsrx
  linkindex: 5
  node_count: 1
  prefix:
    ipv4: 172.16.4.0/24
  type: stub
- bridge: input_6
  interfaces:
  - ipv4: 172.31.0.1/24
    ipv6: 2008:db8:1::1/64
    node: c_nxos
  - ipv4: 172.31.0.2/24
    ipv6: 2008:db8:1::2/64
    node: c_csr
  - ipv4: 172.31.0.3/24
    ipv6: 2008:db8:1::3/64
    node: a_eos
  - ipv4: 172.31.0.4/24
    ipv6: 2008:db8:1::4/64
    node: j_vsrx
  isis:
    metric: 50
  linkindex: 6
  node_count: 4
  prefix:
    ipv4: 172.31.0.0/24
    ipv6: 2008:db8:1::/64
  role: stub
  type: lan
- bridge: input_7
  interfaces:
  - ipv4: 172.16.5.1/24
    node: c_nxos
  - ipv4: 172.16.5.2/24
    node: c_csr
  - ipv4: 172.16.5.3/24
    node: a_eos
  - ipv4: 172.16.5.4/24
    node: j_vsrx
  linkindex: 7
  node_count: 4
  prefix:
    ipv4: 172.16.5.0/24
  role: external
  type: lan
- interfaces:
  - ipv4: true
    node: a_eos
  - ipv4: true
    node: c_nxos
  left:
    ifname: Ethernet5
    ipv4: true
    node: a_eos
  linkindex: 8
  name: a_eos - c_nxos
  node_count: 2
  right:
    ifname: Ethernet1/5
    ipv4: true
    node: c_nxos
  type: p2p
  unnumbered: true
- interfaces:
  - ipv4: true
    node: c_csr
  - ipv4: true
    node: c_nxos
  left:
    ifname: GigabitEthernet6
    ipv4: true
    node: c_csr
  linkindex: 9
  name: c_csr - c_nxos
  node_count: 2
  right:
    ifname: Ethernet1/6
    ipv4: true
    node: c_nxos
  type: p2p
  unnumbered: true
- interfaces:
  - ipv6: true
    node: c_nxos
  - ipv6: true
    node: j_vsrx
  left:
    ifname: Ethernet1/7
    ipv6: true
    node: c_nxos
  linkindex: 10
  name: c_nxos - j_vsrx
  node_count: 2
  prefix:
    ipv6: true
  right:
    ifname: ge-0/0/4
    ipv6: true
    node: j_vsrx
  type: p2p
- interfaces:
  - ipv6: true
    node: a_eos
  - ipv6: true
    node: j_vsrx
  left:
    ifname: Ethernet6
    ipv6: true
    node: a_eos
  linkindex: 11
  name: a_eos - j_vsrx
  node_count: 2
  prefix:
    ipv6: true
  right:
    ifname: ge-0/0/5
    ipv6: true
    node: j_vsrx
  type: p2p
- interfaces:
  - ipv4: true
    node: a_eos
  - ipv4: true
    node: c_csr
  left:
    ifname: Ethernet7
    ipv4: true
    node: a_eos
  linkindex: 12
  name: a_eos - c_csr
  node_count: 2
  right:
    ifname: GigabitEthernet7
    ipv4: true
    node: c_csr
  type: p2p
  unnumbered: true
- interfaces:
  - ipv6: true
    node: c_csr
  - ipv6: true
    node: j_vsrx
  left:
    ifname: GigabitEthernet8
    ipv6: true
    node: c_csr
  linkindex: 13
  name: c_csr - j_vsrx
  node_count: 2
  prefix:
    ipv6: true
  right:
    ifname: ge-0/0/6
    ipv6: true
    node: j_vsrx
  type: p2p
- interfaces:
  - node: a_eos
  - node: c_nxos
  left:
    ifname: Ethernet8
    node: a_eos
  linkindex: 14
  name: L2-only link to test removal of isis context
  node_count: 2
  prefix:
    ipv4: false
  right:
    ifname: Ethernet1/8
    node: c_nxos
  type: p2p
module:
- isis
name: input
nodes:
  a_eos:
    box: arista/veos
    device: eos
    id: 3
    interfaces:
    - bridge: input_1
      ifindex: 1
      ifname: Ethernet1
      ipv4: 172.16.0.3/24
      isis:
        metric: 30
        passive: false
      linkindex: 1
      name: a_eos -> [c_nxos,c_csr,j_vsrx]
      neighbors:
      - ifname: Ethernet1/1
        ipv4: 172.16.0.1/24
        node: c_nxos
      - ifname: GigabitEthernet2
        ipv4: 172.16.0.2/24
        node: c_csr
      - ifname: ge-0/0/0
        ipv4: 172.16.0.4/24
        node: j_vsrx
      type: lan
    - bridge: input_4
      ifindex: 2
      ifname: Ethernet2
      ipv4: 172.16.3.3/24
      isis:
        passive: true
      linkindex: 4
      neighbors: []
      type: stub
    - bridge: input_6
      ifindex: 3
      ifname: Ethernet3
      ipv4: 172.31.0.3/24
      ipv6: 2008:db8:1::3/64
      isis:
        metric: 50
        passive: true
      linkindex: 6
      name: a_eos -> [c_nxos,c_csr,j_vsrx]
      neighbors:
      - ifname: Ethernet1/3
        ipv4: 172.31.0.1/24
        ipv6: 2008:db8:1::1/64
        node: c_nxos
      - ifname: GigabitEthernet4
        ipv4: 172.31.0.2/24
        ipv6: 2008:db8:1::2/64
        node: c_csr
      - ifname: ge-0/0/2
        ipv4: 172.31.0.4/24
        ipv6: 2008:db8:1::4/64
        node: j_vsrx
      role: stub
      type: lan
    - bridge: input_7
      ifindex: 4
      ifname: Ethernet4
      ipv4: 172.16.5.3/24
      linkindex: 7
      name: a_eos -> [c_nxos,c_csr,j_vsrx]
      neighbors:
      - ifname: Ethernet1/4
        ipv4: 172.16.5.1/24
        node: c_nxos
      - ifname: GigabitEthernet5
        ipv4: 172.16.5.2/24
        node: c_csr
      - ifname: ge-0/0/3
        ipv4: 172.16.5.4/24
        node: j_vsrx
      role: external
      type: lan
    - ifindex: 5
      ifname: Ethernet5
      ipv4: true
      isis:
        network_type: point-to-point
        passive: false
      linkindex: 8
      name: a_eos -> c_nxos
      neighbors:
      - ifname: Ethernet1/5
        ipv4: true
        node: c_nxos
      remote_id: 1
      remote_ifindex: 5
      type: p2p
      unnumbered: true
    - ifindex: 6
      ifname: Ethernet6
      ipv6: true
      isis:
        network_type: point-to-point
        passive: false
      linkindex: 11
      name: a_eos -> j_vsrx
      neighbors:
      - ifname: ge-0/0/5
        ipv6: true
        node: j_vsrx
      remote_id: 4
      remote_ifindex: 5
      type: p2p
    - ifindex: 7
      ifname: Ethernet7
      ipv4: true
      isis:
        network_type: point-to-point
        passive: false
      linkindex: 12
      name: a_eos -> c_csr
      neighbors:
      - ifname: GigabitEthernet7
        ipv4: true
        node: c_csr
      remote_id: 2
      remote_ifindex: 7
      type: p2p
      unnumbered: true
    - ifindex: 8
      ifname: Ethernet8
      linkindex: 14
      name: L2-only link to test removal of isis context
      neighbors:
      - ifname: Ethernet1/8
        node: c_nxos
      remote_id: 1
      remote_ifindex: 8
      type: p2p
    isis:
      area: 49.0002
      ipv4: true
      ipv6: true
      type: level-2
    loopback:
      ipv4: 10.0.0.3/32
    mgmt:
      ifname: Management1
      ipv4: 192.168.121.103
      mac: 08-4F-A9-00-00-03
    module:
    - isis
    name: a_eos
  c_csr:
    box: cisco/csr1000v
    device: csr
    id: 2
    interfaces:
    - bridge: input_1
      ifindex: 2
      ifname: GigabitEthernet2
      ipv4: 172.16.0.2/24
      isis:
        metric: 20
        passive: false
      linkindex: 1
      name: c_csr -> [c_nxos,a_eos,j_vsrx]
      neighbors:
      - ifname: Ethernet1/1
        ipv4: 172.16.0.1/24
        node: c_nxos
      - ifname: Ethernet1
        ipv4: 172.16.0.3/24
        node: a_eos
      - ifname: ge-0/0/0
        ipv4: 172.16.0.4/24
        node: j_vsrx
      type: lan
    - bridge: input_3
      ifindex: 3
      ifname: GigabitEthernet3
      ipv4: 172.16.2.2/24
      isis:
        passive: true
      linkindex: 3
      neighbors: []
      type: stub
    - bridge: input_6
      ifindex: 4
      ifname: GigabitEthernet4
      ipv4: 172.31.0.2/24
      ipv6: 2008:db8:1::2/64
      isis:
        metric: 50
        passive: true
      linkindex: 6
      name: c_csr -> [c_nxos,a_eos,j_vsrx]
      neighbors:
      - ifname: Ethernet1/3
        ipv4: 172.31.0.1/24
        ipv6: 2008:db8:1::1/64
        node: c_nxos
      - ifname: Ethernet3
        ipv4: 172.31.0.3/24
        ipv6: 2008:db8:1::3/64
        node: a_eos
      - ifname: ge-0/0/2
        ipv4: 172.31.0.4/24
        ipv6: 2008:db8:1::4/64
        node: j_vsrx
      role: stub
      type: lan
    - bridge: input_7
      ifindex: 5
      ifname: GigabitEthernet5
      ipv4: 172.16.5.2/24
      linkindex: 7
      name: c_csr -> [c_nxos,a_eos,j_vsrx]
      neighbors:
      - ifname: Ethernet1/4
        ipv4: 172.16.5.1/24
        node: c_nxos
      - ifname: Ethernet4
        ipv4: 172.16.5.3/24
        node: a_eos
      - ifname: ge-0/0/3
        ipv4: 172.16.5.4/24
        node: j_vsrx
      role: external
      type: lan
    - ifindex: 6
      ifname: GigabitEthernet6
      ipv4: true
      isis:
        network_type: point-to-point
        passive: false
      linkindex: 9
      name: c_csr -> c_nxos
      neighbors:
      - ifname: Ethernet1/6
        ipv4: true
        node: c_nxos
      remote_id: 1
      remote_ifindex: 6
      type: p2p
      unnumbered: true
    - ifindex: 7
      ifname: GigabitEthernet7
      ipv4: true
      isis:
        network_type: point-to-point
        passive: false
      linkindex: 12
      name: c_csr -> a_eos
      neighbors:
      - ifname: Ethernet7
        ipv4: true
        node: a_eos
      remote_id: 3
      remote_ifindex: 7
      type: p2p
      unnumbered: true
    - ifindex: 8
      ifname: GigabitEthernet8
      ipv6: true
      isis:
        network_type: point-to-point
        passive: false
      linkindex: 13
      name: c_csr -> j_vsrx
      neighbors:
      - ifname: ge-0/0/6
        ipv6: true
        node: j_vsrx
      remote_id: 4
      remote_ifindex: 6
      type: p2p
    isis:
      area: 49.0003
      ipv4: true
      ipv6: true
      type: level-2
    loopback:
      ipv4: 10.0.0.2/32
    mgmt:
      ifname: GigabitEthernet1
      ipv4: 192.168.121.102
      mac: 08-4F-A9-00-00-02
    module:
    - isis
    name: c_csr
  c_nxos:
    box: cisco/nexus9300v
    device: nxos
    id: 1
    interfaces:
    - bridge: input_1
      ifindex: 1
      ifname: Ethernet1/1
      ipv4: 172.16.0.1/24
      isis:
        metric: 10
        passive: false
      linkindex: 1
      name: c_nxos -> [c_csr,a_eos,j_vsrx]
      neighbors:
      - ifname: GigabitEthernet2
        ipv4: 172.16.0.2/24
        node: c_csr
      - ifname: Ethernet1
        ipv4: 172.16.0.3/24
        node: a_eos
      - ifname: ge-0/0/0
        ipv4: 172.16.0.4/24
        node: j_vsrx
      type: lan
    - bridge: input_2
      ifindex: 2
      ifname: Ethernet1/2
      ipv4: 172.16.1.1/24
      isis:
        passive: true
      linkindex: 2
      neighbors: []
      type: stub
    - bridge: input_6
      ifindex: 3
      ifname: Ethernet1/3
      ipv4: 172.31.0.1/24
      ipv6: 2008:db8:1::1/64
      isis:
        metric: 50
        passive: true
      linkindex: 6
      name: c_nxos -> [c_csr,a_eos,j_vsrx]
      neighbors:
      - ifname: GigabitEthernet4
        ipv4: 172.31.0.2/24
        ipv6: 2008:db8:1::2/64
        node: c_csr
      - ifname: Ethernet3
        ipv4: 172.31.0.3/24
        ipv6: 2008:db8:1::3/64
        node: a_eos
      - ifname: ge-0/0/2
        ipv4: 172.31.0.4/24
        ipv6: 2008:db8:1::4/64
        node: j_vsrx
      role: stub
      type: lan
    - bridge: input_7
      ifindex: 4
      ifname: Ethernet1/4
      ipv4: 172.16.5.1/24
      linkindex: 7
      name: c_nxos -> [c_csr,a_eos,j_vsrx]
      neighbors:
      - ifname: GigabitEthernet5
        ipv4: 172.16.5.2/24
        node: c_csr
      - ifname: Ethernet4
        ipv4: 172.16.5.3/24
        node: a_eos
      - ifname: ge-0/0/3
        ipv4: 172.16.5.4/24
        node: j_vsrx
      role: external
      type: lan
    - ifindex: 5
      ifname: Ethernet1/5
      ipv4: true
      isis:
        network_type: point-to-point
        passive: false
      linkindex: 8
      name: c_nxos -> a_eos
      neighbors:
      - ifname: Ethernet5
        ipv4: true
        node: a_eos
      remote_id: 3
      remote_ifindex: 5
      type: p2p
      unnumbered: true
    - ifindex: 6
      ifname: Ethernet1/6
      ipv4: true
      isis:
        network_type: point-to-point
        passive: false
      linkindex: 9
      name: c_nxos -> c_csr
      neighbors:
      - ifname: GigabitEthernet6
        ipv4: true
        node: c_csr
      remote_id: 2
      remote_ifindex: 6
      type: p2p
      unnumbered: true
    - ifindex: 7
      ifname: Ethernet1/7
      ipv6: true
      isis:
        network_type: point-to-point
        passive: false
      linkindex: 10
      name: c_nxos -> j_vsrx
      neighbors:
      - ifname: ge-0/0/4
        ipv6: true
        node: j_vsrx
      remote_id: 4
      remote_ifindex: 4
      type: p2p
    - ifindex: 8
      ifname: Ethernet1/8
      linkindex: 14
      name: L2-only link to test removal of isis context
      neighbors:
      - ifname: Ethernet8
        node: a_eos
      remote_id: 3
      remote_ifindex: 8
      type: p2p
    isis:
      area: 49.0002
      ipv4: true
      ipv6: true
      net: 49.0003.0000.0000.1234.00
      type: level-2
    loopback:
      ipv4: 10.0.0.1/32
    mgmt:
      ifname: mgmt0
      ipv4: 192.168.121.101
      mac: 08-4F-A9-00-00-01
    module:
    - isis
    name: c_nxos
  j_vsrx:
    box: juniper/vsrx3
    device: vsrx
    id: 4
    interfaces:
    - bridge: input_1
      ifindex: 0
      ifname: ge-0/0/0
      ipv4: 172.16.0.4/24
      isis:
        metric: 50
        passive: false
      linkindex: 1
      name: j_vsrx -> [c_nxos,c_csr,a_eos]
      neighbors:
      - ifname: Ethernet1/1
        ipv4: 172.16.0.1/24
        node: c_nxos
      - ifname: GigabitEthernet2
        ipv4: 172.16.0.2/24
        node: c_csr
      - ifname: Ethernet1
        ipv4: 172.16.0.3/24
        node: a_eos
      type: lan
    - bridge: input_5
      ifindex: 1
      ifname: ge-0/0/1
      ipv4: 172.16.4.4/24
      isis:
        passive: true
      linkindex: 5
      neighbors: []
      type: stub
    - bridge: input_6
      ifindex: 2
      ifname: ge-0/0/2
      ipv4: 172.31.0.4/24
      ipv6: 2008:db8:1::4/64
      isis:
        metric: 50
        passive: true
      linkindex: 6
      name: j_vsrx -> [c_nxos,c_csr,a_eos]
      neighbors:
      - ifname: Ethernet1/3
        ipv4: 172.31.0.1/24
        ipv6: 2008:db8:1::1/64
        node: c_nxos
      - ifname: GigabitEthernet4
        ipv4: 172.31.0.2/24
        ipv6: 2008:db8:1::2/64
        node: c_csr
      - ifname: Ethernet3
        ipv4: 172.31.0.3/24
        ipv6: 2008:db8:1::3/64
        node: a_eos
      role: stub
      type: lan
    - bridge: input_7
      ifindex: 3
      ifname: ge-0/0/3
      ipv4: 172.16.5.4/24
      linkindex: 7
      name: j_vsrx -> [c_nxos,c_csr,a_eos]
      neighbors:
      - ifname: Ethernet1/4
        ipv4: 172.16.5.1/24
        node: c_nxos
      - ifname: GigabitEthernet5
        ipv4: 172.16.5.2/24
        node: c_csr
      - ifname: Ethernet4
        ipv4: 172.16.5.3/24
        node: a_eos
      role: external
      type: lan
    - ifindex: 4
      ifname: ge-0/0/4
      ipv6: true
      isis:
        network_type: point-to-point
        passive: false
      linkindex: 10
      name: j_vsrx -> c_nxos
      neighbors:
      - ifname: Ethernet1/7
        ipv6: true
        node: c_nxos
      remote_id: 1
      remote_ifindex: 7
      type: p2p
    - ifindex: 5
      ifname: ge-0/0/5
      ipv6: true
      isis:
        network_type: point-to-point
        passive: false
      linkindex: 11
      name: j_vsrx -> a_eos
      neighbors:
      - ifname: Ethernet6
        ipv6: true
        node: a_eos
      remote_id: 3
      remote_ifindex: 6
      type: p2p
    - ifindex: 6
      ifname: ge-0/0/6
      ipv6: true
      isis:
        network_type: point-to-point
        passive: false
      linkindex: 13
      name: j_vsrx -> c_csr
      neighbors:
      - ifname: GigabitEthernet8
        ipv6: true
        node: c_csr
      remote_id: 2
      remote_ifindex: 8
      type: p2p
    isis:
      area: 49.0002
      ipv4: true
      ipv6: true
      type: level-2
    loopback:
      ipv4: 10.0.0.4/32
    mgmt:
      ifname: fxp0
      ipv4: 192.168.121.104
      mac: 08-4F-A9-00-00-04
    module:
    - isis
    name: j_vsrx
provider: libvirt
